src/core/stream/adaptation/types.ts

```diff
@@ -4,14 +4,15 @@ import type {
   IAudioTrackSwitchingMode,
   IVideoTrackSwitchingMode,
 } from "../../../public_types";
+import type { IRange } from "../../../utils/ranges";
 import type { IReadOnlySharedReference } from "../../../utils/reference";
 import type { IRepresentationEstimator } from "../../adaptive";
 import type { SegmentQueueCreator } from "../../fetchers";
 import type { IBufferType, SegmentSink } from "../../segment_sinks";
+import type { IPeriodStreamPlaybackObservation } from "../period";
```

Comment:
To me the PeriodStream depends on the AdaptationStream, not the other way around

Reply:
The thing is that they don't overlap well, `IAdaptationStreamPlaybackObservation` has `bufferGap` attribute in addition of the other attributes that as `IPeriodStreamPlaybackObservation`.
 But they both defines the `buffered` attribute in a different way.
 
 I will just define them as both types with no `extends` nor `Omit`

Reply:
We could also use different property names to explicit that difference